test/unit/gateways/worldpay_test.rb

```diff
@@ -145,11 +145,36 @@ def setup
     }
   end
 
+  def test_successfully_attaching_payment_type
+    [@credit_card, @nt_credit_card].each do |pm|
+      payment = @gateway.send(:payment_details, pm)[:payment_type]
+      if pm.respond_to?(:source)
```

Comment:
I think it's better to split these into separate tests to keep the passing conditions clear. This test may result in a false positive later on if, for example, a payment method contains a `source` field and it shouldn't.

Reply:
It wouldn't be possible for CreditCard to have the `source` field but I can still separate.